### PR TITLE
update make-bm-worker tool to include bootMacAddress

### PR DESCRIPTION
**Pull request by rlopez133**
Currently the make-bm-worker tool does not include the bootMACAddress that is required in order to specify the provisioning NIC when adding a worker node to your KNI deployment. This code includes the bootMACAddress as a field that needs to be filled out when using the bm-worker tool.

### CODE REVIEW

File: cmd/make-bm-worker/main.go [hunk 1]
```diff
@@ -39,13 +40,18 @@ func main() {
 		fmt.Fprintf(os.Stderr, "Missing -address argument\n")
 		os.Exit(1)
 	}
+	if *macAddress == "" {
```
Comment (dhellmann) [inline cmd/make-bm-worker/main.go]: The MAC address is not always required. In fact, when inspection is available it won't be required for any bare metal systems. So I think this option should stay optional.

Comment (russellb) [reply cmd/make-bm-worker/main.go]: It's discovery that makes this optional, not introspection, right?

Comment (rlopez133) [reply cmd/make-bm-worker/main.go]: @dhellmann - when troubleshooting adding a worker, this is one thing @russellb mentioned to me that I was missing within my yaml file. The MAC address of this host for its interface on the provisioning network. I guess I'm unsure then what I would or would not need to include it.

Comment (dhellmann) [reply cmd/make-bm-worker/main.go]: > It's discovery that makes this optional, not introspection, right?

Introspection happens on discovered hosts, but it now also happens on registered hosts. My understanding from @dtantsur and @juliakreger is that if the user enters the BMC settings and we power it on and do inspection, the user does not have to enter the MAC.

Comment (dhellmann) [reply cmd/make-bm-worker/main.go]: > @dhellmann - when troubleshooting adding a worker, this is one thing @russellb mentioned to me that I was missing within my yaml file. The MAC address of this host for its interface on the provisioning network. I guess I'm unsure then what I would or would not need to include it.

My understanding is that you only have to enter the value for VMs or in cases where inspection isn't running (so for older versions of the baremetal-operator).

File: cmd/make-bm-worker/main.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ func main() {
 	var password = flag.String("password", "", "password for BMC")
 	var bmcAddress = flag.String("address", "", "address URL for BMC")
 	var hardwareProfile = flag.String("hardwareprofile", "", "hardwareProfile to be used")
+	var macAddress = flag.String("bootmacaddress", "", "bootMACAddress to be used")
```
Comment (russellb) [inline cmd/make-bm-worker/main.go]: There's another version of this tool in dev-scripts, which includes "-boot-mac".  Can we use the same name to match?

Comment (rlopez133) [reply cmd/make-bm-worker/main.go]: Done.

File: cmd/make-bm-worker/main.go (hunk 1 above)
Comment (russellb) [reply cmd/make-bm-worker/main.go]: seems like the MAC is still needed to be able to configure the PXE environment so it'll boot into the IPA image for introspection?

anyway, seems fine to make this optional in the script

Comment (rlopez133) [reply cmd/make-bm-worker/main.go]: @dhellmann - I went ahead and removed bootmac from being required to being optional.

**Comment by russellb** [comment]
Feedback seems addressed, and works for me!

**Comment by russellb** [comment]
One last thing ... can you squash these commits down to 1 commit?  You'd do something like `git rebase -i origin/master` and then "squash" all but one commit.

File: cmd/make-bm-worker/main.go (hunk 1 above)
Comment (dhellmann) [reply cmd/make-bm-worker/main.go]: > seems like the MAC is still needed to be able to configure the PXE environment so it'll boot into the IPA image for introspection?
> 
> anyway, seems fine to make this optional in the script

The DHCP server is configured to boot the IPA image for any MAC it doesn't recognize. That allows discovery to work, and collect the MAC address(es) of the host. Those values can then be used to reconfigure the DHCP server for that one host to be given more specific instructions (it is possible to return a PXE response to boot directly from the local disk, for example), although I don't know if we're actually doing that.

**Comment by rlopez133** [comment]
Feel

**Pull request closed without merging**